File: frontend/src/app/_model/company.ts
/**
 * Created by Joinsu on 2017-05-12.
 */
export class Company{
  private id: number;
  private name: string;
  private country: string;
  private street: string;
  private city: string;
  private province: string;
  private postal: string;
  private phone: string;
  private email: string;
  private website: string;
  private fax: string;
  private gstNo: string;

  getId(): number {
    return this.id;
  }

  setId( value: number) {
    this.id = value;
  }

  getName(): string {
    return this.name;
  }

  setName(value: string) {
    this.name = value;
  }

  getCountry(): string {
    return this.country;
  }

  setCountry(value: string) {
    this.country = value;
  }

  getStreet(): string {
    return this.street;
  }

  setStreet(value: string) {
    this.street = value;
  }

  getCity(): string {
    return this.city;
  }

  setCity(value: string) {
    this.city = value;
  }

  getProvince(): string {
    return this.province;
  }

  setProvince(value: string) {
    this.province = value;
  }

  getPostal(): string {
    return this.postal;
  }

  setPostal(value: string) {
    this.postal = value;
  }

  getPhone(): string {
    return this.phone;
  }

  setPhone(value: string) {
    this.phone = value;
  }

  getEmail(): string {
    return this.email;
  }

  setEmail(value: string) {
    this.email = value;
  }

  getWebsite(): string {
    return this.website;
  }

  setWebsite(value: string) {
    this.website = value;
  }

  getFax(): string {
    return this.fax;
  }

  setFax(value: string) {
    this.fax = value;
  }

  getGstNo(): string {
    return this.gstNo;
  }

  setGstNo(value: string) {
    this.gstNo = value;
  }

  get fullAddress(){
    let space:string = ', ';
    return this.street + space + this.city + space + this.province + space + this.country + ' ' + this.postal;
  }

  setDataFromResponse(responseData: any){
    this.setId( responseData.id);
    this.setName( responseData.name);
    this.setCountry( responseData.country);
    this.setProvince( responseData.province);
    this.setCity( responseData.city);
    this.setStreet( responseData.street);
    this.setPostal( responseData.postal);
    this.setPhone( responseData.phone);
    this.setFax( responseData.fax);
    this.setEmail( responseData.email);
    this.setWebsite( responseData.website);
    this.setGstNo( responseData.gstNo);
  }
}

File: src/main/java/ca/joyfactory/coims/domain/Company.java
package ca.joyfactory.coims.domain;

import ca.joyfactory.coims.common.CommonUtil;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Created by Joinsu on 2017-09-27.
 */
@Entity
public class Company {

    @Transient
    public static long DEFAULT_ID = -1L;

    @Id
    @Column(name = "COMPANY_ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToMany( mappedBy = "company", orphanRemoval = true)
    // TODO: 무한루프 없애기 위해 일단은 조치 확인 필요
//    @JsonIgnoreProperties( "company")
    @JsonIgnore
    private List<User> users = new ArrayList<User>();

    @OneToMany( mappedBy = "company", orphanRemoval = true)
//    @JsonIgnoreProperties( "company")
    @JsonIgnore
    private List<CaseTypeFee> caseTypeFees = new ArrayList<>();

    @Column(length = 50)
    private String name;
    @Column(length = 10)
    private String country;
    @Column(length = 50)
    private String street;
    @Column(length = 20)
    private String city;
    @Column(length = 10)
    private String province;
    @Column(length = 10)
    private String postal;
    @Column(length = 13)
    private String phone;
    @Column(length = 50)
    private String email;
    @Column(length = 20)
    private String website;
    @Column(length = 13)
    private String fax;
    @Column(length = 15)
    private Integer gstNo;

    private Date createdDate;
    private Date lastModifiedDate;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public List<User> getUsers() {
        return users;
    }

    public void setUsers(List<User> users) {
        this.users = users;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getPostal() {

        return this.postal;
    }

    public void setPostal(String postal) {
        this.postal = postal;
    }

    public String getPhone() {
        return this.phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getWebsite() {
        return this.website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public String getFax() {

        return this.fax;
    }

    public void setFax(String fax) {
        this.fax = fax;
    }

    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }

    public Date getLastModifiedDate() {
        return lastModifiedDate;
    }

    public void setLastModifiedDate(Date lastModifiedDate) {
        this.lastModifiedDate = lastModifiedDate;
    }

    public String getFullAddress(){
        String divide = ", ";
        return this.getStreet() + divide + this.getCity() + divide + this.getProvince() + divide + this.getCountry() + divide + this.getPostal();
    }

    public List<CaseTypeFee> getCaseTypeFees() {
        return caseTypeFees;
    }

    public void setCaseTypeFees(List<CaseTypeFee> caseTypeFees) {
        this.caseTypeFees = caseTypeFees;
    }

    public Integer getGstNo() {
        return gstNo;
    }

    public void setGstNo(Integer gstNo) {
        this.gstNo = gstNo;
    }
}
